src/xunit.analyzers/X1000/MemberDataShouldReferenceValidMember.cs

```diff
@@ -234,7 +239,12 @@ public class MemberDataShouldReferenceValidMember : XunitDiagnosticAnalyzer
 		if (initializer is null)
 			return new List<ExpressionSyntax> { argumentExpression };
 
-		return initializer.Expressions.ToList();
+		// In the special case where the argument is an object[], treat like params
+		var type = semanticModel.GetTypeInfo(argumentExpression).Type;
+		if (type is IArrayTypeSymbol arrayType && arrayType.ElementType.SpecialType == SpecialType.System_Object)
+			return initializer.Expressions.ToList();
+
+		return new List<ExpressionSyntax> { argumentExpression };
```

Comment:
@JamesTerwilliger could this be shortcutted by inverting the `if` on 244 and executing the `if ` between line 231 and 232? then in this case, it does not have to determine `initializer`. Thoughts?

Reply:
@viceroypenguin I had it this way thinking that initializer being not null means that it is an array type, and that the semantic model call is likely more expensive. Said differently, the initializer being null is the short-circuit that I want so that the (I believe) expensive call is only done if need be.

Reply:
Good reasoning; and I'll trust your experience with this code, not having enough experience with touching the `semanticModel` in analyzers. LGTM